test/unit-tests/api-gateway/validation/oauth2/oauthTokenValidatorTest.lua

```diff
@@ -0,0 +1,52 @@
+-- Copyright (c) 2018 Adobe Systems Incorporated. All rights reserved.
+--
+--   Permission is hereby granted, free of charge, to any person obtaining a
+--   copy of this software and associated documentation files (the "Software"),
+--   to deal in the Software without restriction, including without limitation
+--   the rights to use, copy, modify, merge, publish, distribute, sublicense,
+--   and/or sell copies of the Software, and to permit persons to whom the
+--   Software is furnished to do so, subject to the following conditions:
+--
+--   The above copyright notice and this permission notice shall be included in
+--   all copies or substantial portions of the Software.
+--
+--   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+--   IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+--   FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+--   AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+--   LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+--   FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+--   DEALINGS IN THE SOFTWARE.
+
+local cjson = require "cjson"
+local safeCjson = require "cjson.safe"
+
+local BaseValidatorMock = mock("api-gateway.validation.validator", {
```

Comment:
Beeing in the same package you sould not need this - you can let it make it's inheritance

Reply:
@atrifan, removing that line throws the following error: 
``` checkResponseFromAuth: should return false for an invalid json ----- FAIL 
...l/share/lua/5.1/api-gateway/validation/validator.lua:36: attempt to index field 'config' (a nil value) in ...ateway/validation/oauth2/oauthTokenValidatorTest.lua : 24 ```